cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -72,307 +78,383 @@ def __init__(
         """
         self._api_key = api_key
 
-        assert (
-            max_concurrent_requests < _MAX_CONCURRENT_TLM_REQUESTS_LIMIT
-        ), f"max_concurrent_requests must be less than {_MAX_CONCURRENT_TLM_REQUESTS_LIMIT}"
-
         if quality_preset not in _VALID_TLM_QUALITY_PRESETS:
-            raise ValueError(
+            raise ValidationError(
                 f"Invalid quality preset {quality_preset} -- must be one of {_VALID_TLM_QUALITY_PRESETS}"
             )
 
+        if options is not None:
+            validate_tlm_options(options)
+
+        if timeout is not None and not (isinstance(timeout, int) or isinstance(timeout, float)):
+            raise ValidationError("timeout must be a integer or float value")
+
+        if verbose is not None and not isinstance(verbose, bool):
+            raise ValidationError("timeout must be a boolean value")
+
+        is_notebook_flag = is_notebook()
+
         self._quality_preset = quality_preset
+        self._options = options
+        self._timeout = timeout if timeout is not None and timeout > 0 else None
+        self._verbose = verbose if verbose is not None else is_notebook_flag
 
-        if is_notebook():
+        if is_notebook_flag:
             import nest_asyncio
 
             nest_asyncio.apply()
 
+        # TODO: figure out this how to compute appropriate max_concurrent_requests
+        max_concurrent_requests = _DEFAULT_MAX_CONCURRENT_TLM_REQUESTS
+
         self._event_loop = asyncio.get_event_loop()
         self._query_semaphore = asyncio.Semaphore(max_concurrent_requests)
 
     def _batch_prompt(
         self,
-        prompts: List[str],
-        options: Union[None, TLMOptions, List[Union[TLMOptions, None]]] = None,
-        timeout: Optional[float] = None,
-        retries: int = 1,
+        prompts: Sequence[str],
     ) -> List[TLMResponse]:
         """Run batch of TLM prompts.
 
         Args:
             prompts (List[str]): list of prompts to run
-            options (None | TLMOptions | List[TLMOptions  |  None], optional): list of options (or instance of options) to pass to prompt method. Defaults to None.
-            timeout (Optional[float], optional): timeout (in seconds) to run all prompts. Defaults to None.
-            retries (int): number of retries to attempt for each individual prompt in case of error. Defaults to 1.
 
         Returns:
             List[TLMResponse]: TLM responses for each prompt (in supplied order)
         """
-        if isinstance(options, list):
-            options_collection = options
-        else:
-            options = cast(Union[None, TLMOptions], options)
-            options_collection = [options for _ in prompts]
+        tlm_responses = self._event_loop.run_until_complete(
+            self._batch_async([self._prompt_async(prompt) for prompt in prompts])
+        )
+
+        return cast(List[TLMResponse], tlm_responses)
+
+    def _batch_prompt_with_handled_errors(
```

Comment:
It might be a little repetitive to define this function (could potentially combine with `_batch_prompt` with a `handle_errors` flag), but I was getting way too many mypy errors (which I wasn't sure how to handle/if I could ignore) so decided to do it separately.